src/main/java/org/jabref/gui/maintable/MainTableHeaderRightClickMenu.java

```diff
@@ -0,0 +1,80 @@
+package org.jabref.gui.maintable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.RadioMenuItem;
+import javafx.scene.control.SeparatorMenuItem;
+import javafx.scene.control.TableColumn;
+import javafx.scene.layout.StackPane;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.LibraryTab;
+import org.jabref.gui.maintable.columns.MainTableColumn;
+import org.jabref.gui.preferences.PreferencesDialogView;
+import org.jabref.logic.l10n.Localization;
+
+public class MainTableHeaderRightClickMenu extends ContextMenu {
+    static MainTable mT = null;
+
+    public void show(MainTable mainTable, LibraryTab libraryTab, DialogService dialogService) {
+        System.out.println("show()");
+        mT = mainTable;
+        mainTable.setOnContextMenuRequested(clickEvent -> {
+
+            // Click on the tableColumns
+            if (!(clickEvent.getTarget() instanceof StackPane)) {
+                System.out.println("Click on the tableColumns");
+                // Create radioMenuItemList from tableColumnList
```

Comment:
Please avoid code-describing comments, instead try to write self-explaining code and put your ratilonale into a comment.